protoc-gen-elm/go_tests/testdata/map_entry/input/map_entry.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+message Bar {
+  bool field = 1;
+}
+
+message Foo {
+    map<string, Bar> stringToBars = 8;
+    map<string, string> stringToStrings = 7;
```

Comment:
So what happens for maps with non-string keys? Are they just serialized as lists as before? Should we add a test case anyways?

Reply:
I went back and put the guard in `main.go` back to only allowing `string` keys:
```golang
if keyField.GetType() != descriptor.FieldDescriptorProto_TYPE_STRING  {
    return fmt.Errorf("map key must have type `string`")
}
```

I originally went down the road of supporting `Int`, but I don't have a use-case that needs them yet, and it would require a different decoder, because `Decode.dict` is written with only `String` keys in mind, and doesn't have a `comparable` version.  At some point, I may add a separate decoder for `Dict Int`'s + add this in.